packages/dds/tree/src/test/shared-tree/undo.spec.ts

```diff
@@ -367,4 +371,45 @@ describe("Undo and redo", () => {
 		expectJsonTree(tree, ["B", "C"]);
 		unsubscribe();
 	});
+
+	it("can undo/redo a merge", () => {
+		const tree = makeTreeFromJsonSequence(["A", "B"]);
+
+		const { undoStack, redoStack, unsubscribe } = createTestUndoRedoStacks(tree.events);
+		const branch = tree.branch();
+		branch.editor.sequenceField(rootField).insert(2, singleJsonCursor("C"));
+		branch.editor.sequenceField(rootField).remove(0, 1);
+		tree.merge(branch);
+
+		expectJsonTree(tree, ["B", "C"]);
+		undoStack.pop()?.revert();
+		expectJsonTree(tree, ["A", "B", "C"]);
+		undoStack.pop()?.revert();
+		expectJsonTree(tree, ["A", "B"]);
+		redoStack.pop()?.revert();
+		expectJsonTree(tree, ["A", "B", "C"]);
+		redoStack.pop()?.revert();
+		expectJsonTree(tree, ["B", "C"]);
+		unsubscribe();
+	});
+
+	it("can undo multiple merges", () => {
+		const tree = makeTreeFromJsonSequence(["A", "B"]);
+
+		const { undoStack, unsubscribe } = createTestUndoRedoStacks(tree.events);
+
+		const branch = tree.branch();
+
+		branch.editor.sequenceField(rootField).insert(2, singleJsonCursor("C"));
+		tree.merge(branch, false);
+		expectJsonTree(tree, ["A", "B", "C"]);
+		undoStack.pop()?.revert();
+
+		branch.editor.sequenceField(rootField).insert(2, singleJsonCursor("C"));
+		tree.merge(branch);
+		expectJsonTree(tree, ["A", "B", "C"]);
+		undoStack.pop()?.revert();
```

Comment:
This test is chiefly concerned about the state after this `revert` call, so it would be good to validate it. (Would be nice to do the same for line 406 although that's less the focus)

Reply:
Agreed, thanks!